src/attributes.jl

```diff
@@ -527,13 +578,35 @@ It is guaranteed to be equivalent but not necessarily identical to the function
 """
 struct ConstraintFunction <: AbstractConstraintAttribute end
 
+function hidden_set!(model::ModelLike, ::ConstraintFunction, constraint_index::ConstraintIndex, func::AbstractFunction)
+    if func_type(constraint_index) != typeof(func)
```

Comment:
This can be done with multiple dispatch,
`hidden_set!(..., ::ConstraintFunction, ::CI{F, S}, ::F)` -> same constraint set
`hidden_set!(..., ::ConstraintFunction, ::CI, ::AbstractFunction)` -> different constraint set

Reply:
Oh yeah good point. This is a hold-over from before